src/main/java/io/vinyldns/java/model/batch/ChangeInput.java

```diff
@@ -13,9 +13,44 @@
  */
 package io.vinyldns.java.model.batch;
 
-public interface ChangeInput {
+import io.vinyldns.java.model.record.RecordType;
+import java.util.Objects;
 
-  ChangeInputType getChangeType();
+public class ChangeInput {
+  private ChangeInputType changeType;
+  private String inputName;
+  private RecordType recordType;
 
-  String getInputName();
+  public ChangeInput(ChangeInputType changeType, String inputName, RecordType recordType) {
+    this.changeType = changeType;
+    this.inputName = inputName;
+    this.recordType = recordType;
```

Comment:
Since we're using `gson.toJson` to serialized the request into JSON, this should be `type` to match what the API is expecting. Otherwise, would need to figure out how to set this field as `type` when it gets serialized into JSON.

Reply:
Yeah good point, hit me on my walk home